bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -297,7 +316,7 @@ public void start() {
         if (scheduledFuture != null) {
             scheduledFuture.cancel(false);
         }
-        scheduledFuture = gcExecutor.scheduleAtFixedRate(this, gcWaitTime, gcWaitTime, TimeUnit.MILLISECONDS);
+        scheduledFuture = gcExecutor.scheduleWithFixedDelay(this, gcWaitTime, gcWaitTime, TimeUnit.MILLISECONDS);
```

Comment:
any reason why did you change this to `FixedDelay`?

Reply:
@sijie `FixedDelay` would ensure that fixed amount of time passes between consecutive runs. This will prevent from GC process overburdening the bookie. With `FixedRate`, the duration of GC run is counted as part of the delay time.